### PR TITLE
v2.3.0

### PR DESCRIPTION
**Pull request by alex-miller-0**
**Features:**

* (#480) Adds support for BLS signing and pubkey export
* (#481, #489, #490) Adds support for encrypted data export (starting with BLS private key export via EIP2335)

**Fixes:**

* (#483) Fixes edge case in canonical ABI tuple parser and adds more ABI decoding test vectors

**Misc:**

* (#474) Updates signing tests
* (#477) Adds smart contract Solana tx tests
* (#473) Updates wallet job structs to correspond to updated firmware (only relevant for tests)
* (#478) Adds new ABI decoding vectors
* (#485) Bump version
* (#488) Removes cruft
* (#491) Move ABI decoding lib out of `devDependencies`

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -529,6 +539,13 @@ function getFwVersionConst (v: Buffer): FirmwareConstants {
     };
   }
 
+  // --- V0.17.X ---
+  // V0.17.0 added support for BLS12-381-G1 pubkeys and G2 sigs
+  if (!legacy && gte(v, [0, 17, 0])) {
+    c.getAddressFlags.push(EXTERNAL.GET_ADDR_FLAGS.BLS12_381_G1_PUB);
+    c.iterIdxAllowed = true;
```
Comment (alex-miller-0) [inline src/constants.ts:546]: Cruft

File: src/functions/getAddresses.ts [hunk 2]
```diff
@@ -104,10 +104,13 @@ export const encodeGetAddressesRequest = ({
   const flags = fwConstants.getAddressFlags || [] as any[];
   const isPubkeyOnly =
     flags.indexOf(flag) > -1 &&
-    (flag === EXTERNAL.GET_ADDR_FLAGS.ED25519_PUB ||
-      flag === EXTERNAL.GET_ADDR_FLAGS.SECP256K1_PUB);
+    (
+      flag === EXTERNAL.GET_ADDR_FLAGS.ED25519_PUB ||
+      flag === EXTERNAL.GET_ADDR_FLAGS.SECP256K1_PUB ||
+      flag === EXTERNAL.GET_ADDR_FLAGS.BLS12_381_G1_PUB
+    );
```
Comment (alex-miller-0) [inline src/functions/getAddresses.ts:111]: Can be replaced by a util function

File: src/constants.ts (hunk 1 above)
Comment (alex-miller-0) [reply src/constants.ts]: https://github.com/GridPlus/gridplus-sdk/pull/488

**Review by douglance — APPROVED** [review]

**Comment by paulmillr** [comment]
Keep in mind that bls12 did not went through a security audit. It is kinda good, and fuzzed, but nevertheless.
